internal/backends/linuxkms/renderer/femtovg.rs

```diff
@@ -30,22 +31,17 @@ struct GlContextWrapper {
 }
 
 impl GlContextWrapper {
-    fn new(egl_display: &EglDisplay) -> Result<Self, PlatformError> {
-        let width: std::num::NonZeroU32 = egl_display.size.width.try_into().map_err(|_| {
-            format!(
-                "Attempting to create window surface with an invalid width: {}",
-                egl_display.size.width
-            )
+    fn new(gbm_display: &GbmDisplay) -> Result<Self, PlatformError> {
+        let (width, height) = gbm_display.drm_output.size();
+        let width: std::num::NonZeroU32 = width.try_into().map_err(|_| {
+            format!("Attempting to create window surface with an invalid width: {}", width)
         })?;
-        let height: std::num::NonZeroU32 = egl_display.size.height.try_into().map_err(|_| {
-            format!(
-                "Attempting to create window surface with an invalid height: {}",
-                egl_display.size.height
-            )
+        let height: std::num::NonZeroU32 = height.try_into().map_err(|_| {
+            format!("Attempting to create window surface with an invalid height: {}", height)
         })?;
 
-        let display_handle = egl_display.display_handle().unwrap();
-        let window_handle = egl_display.window_handle().unwrap();
+        let display_handle = gbm_display.display_handle().unwrap();
+        let window_handle = gbm_display.window_handle().unwrap();
```

Comment:
can this really never panic?

Reply:
Heh yes, for once the unwrap is safe, as we always return Ok:

https://github.com/slint-ui/slint/blob/b64b58981d6a481ab2bd73f75b234783fff40a01/internal/backends/linuxkms/display/gbmdisplay.rs#L96